airflow/contrib/kubernetes/kubernetes_request_factory/kubernetes_request_factory.py

```diff
@@ -187,12 +187,17 @@ def extract_image_pull_secrets(pod, req):
                 'name': pull_secret
             } for pull_secret in pod.image_pull_secrets.split(',')]
 
+    @staticmethod
+    def extract_security_context(pod, req):
+        if pod.security_context is None:
+            return
+
+        # Add security_context if not present in spec
+        req['spec']['securityContext'] = req['spec'].get('securityContext', {})
```

Comment:
This line used to be 

```req['spec']['securityContext'] = pod.security_context
```

Now we pull it out of req.sec.securityContext - but I don't think that is ever set anywhere is it? Do we pull "base" values out of pod.security_context anywhere/anymore? Is req['spec']['securityContext']  ever set to anything?

```suggestion
        req['spec'].setdefault('securityContext', {})
```

I've lost all context on this PR and am can't follow through what comes from where.

Reply:
@ashb I understand. 😃 The context is a bit harder to follow now since changes (f4253a29) were merged since this PR was submitted that touched the same area.

Previously the only option that was added from `pod.security_context` was `fsGroup = 65533`, if SSH private key support from f4253a29 was enabled, so we never actually pulled base values from `pod.security_context` (or anywhere else). This PR adds the ability to customize `fsGroup` and `runAsUser` while maintaining compatibility as much as possible with the changes introduced in the aforementioned commit.

Right now these are the only two security context options that are implemented in this PR, but it has been mentioned on slack that it might be beneficial to support the full range since some more locked down clusters might need them. I could add them but `fsGroup` and `runAsUser` are definitely the most common SC options so that's why I focused on them first.

I am committing your suggested change.

Reply:
@ashb Any further questions or requested changes?